src/pathway/GPU-kernel.cuh

```diff
@@ -13,6 +13,10 @@
 #define THREAD_ID (threadIdx.x)
 #define GLOBAL_ID (THREAD_ID + NT * blockIdx.x)
 #define BLOCK_ID  (blockIdx.x)
+#
+#ifndef UINT32_MAX
+#define UINT32_MAX  (0xffffffff)
+#endif
 
```

Comment:
This should be unnecessary on a standard c++.  `stdint.h` should contain them: http://www.cplusplus.com/reference/cstdint/  (included in `src/utils.hpp`).  Could you give them a check?